### ISSUE TITLE
Input gestures dialog: Filter to only show customised entries

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4900]
**Reported by leonarddr on 2015-02-09 18:19**
I wonder whether it could be possible to add a checkbox to the input gestures dialog which allows you to only show the customised gestures, i.e. the ones defined/overwritten in gestures.ini. This way, it would be easier to clean up, change or review customised entries.

**Comment by leonardder** [comment]
The priority of this is quite low. Labelling as feature

**Comment by derekriemer** [comment]
Should we be using feature or enhancement?

**Comment by derekriemer** [comment]
whoops, slip of the finger, aaah

**Comment by derekriemer** [comment]
@feerrenrut p3 or p4?

**Comment by feerrenrut** [comment]
I don't think that we need to provide a priority on this, essentially because it is a feature its unlikely to get worked on outside of a project. Bugs come first.

Before we leave this issue, do we understand the use case here?

I think the most important part of the initial description is the last part:

> easier to clean up, change or review customised entries

Ignoring the suggested implementation for a moment, perhaps we can consider the request as a user story:
"As a user of NVDA, I can differentiate between the built-in, customised, and new gestures listed in the Input Gestures dialog so that it is easier to clean up, change, or review customised entries"

This may be more difficult to deal with (or run into more regularly) than it seems, it would be handy if we knew whether this was a common pain point with the UX.

**Comment by bhavyashah** [comment]
This is a friendly reminder for @leonardder, the original author of this ticket, to kindly consider responding to @feerrenrut's https://github.com/nvaccess/nvda/issues/4900#issuecomment-319254739. Personally speaking, I do not use the Input Gestures dialog extensively, but the requested feature sounds quite justified, at least in theory. How about a checkable dropdown list after the tree view or before the Filter by edit field in the Input Gestures dialog for filtering whether to show default unchanged gestures, functions that are unassigned, and customized gestures, or a combination of these. By default of course, all three should be checked, so that the user has access to all available input gestures up for assignment/reassignment.

**Comment by Qchristensen** [comment]
A user noted today that if you have set custom gestures for an add-on, and then uninstall the add-on, the custom gestures remain.

That perhaps lends weight to the use case for "cleaning up" the gestures list.

**Comment by CyrilleB79** [comment]
In addition to being able to filter custom gestures, the possibility to identify native gestures vs custom gestures is also needed; see #14366 (closed as duplicate) where this is discussed more in depth.
